src/install.js

```diff
@@ -75,21 +109,47 @@ async function downloadSAPUI5(downloadURL, downloadPath) {
 
     response.data.pipe(fs.createWriteStream(zipFile))
     return new Promise((resolve, reject) => {
-      response.data.on('end', () => { resolve(zipFile) })
-      response.data.on('error', () => { reject() })
+      response.data.on('end', () => {
+        resolve(zipFile)
+      })
+      response.data.on('error', () => {
+        reject()
+      })
     })
   } catch (downloadError) {
     throw new Error(`Couldn't download SAPUI5 zip archive from '${downloadURL}'`)
   }
 }
 
 async function extractArchive(archive, targetPath) {
-  console.log('Extracting...')
-  const zip = new AdmZip(archive)
-  zip.extractAllTo(targetPath, true)
+  console.log(`Extracting files to  ${targetPath}`)
+
+  const zip = new StreamZip({
+    file: archive,
+    storeEntries: true,
+  })
+
+
+  zip.on('ready', () => {
+    console.log(`All entries read: ${zip.entriesCount}`)
+
+    const progressBar1 = new ProgressBar('Extracting: [:bar]', {
+
+      total: zip.entriesCount,
+    })
+    zip.on('extract', () => {
+      progressBar1.tick()
+      if (progressBar1.complete) {
+        console.log('SAPUI5 Installed')
+      }
+    })
+    zip.extract(null, targetPath, () => {
+      zip.close()
+    })
+  })
 }
 
-const libDir = path.resolve(`${__dirname}/../lib`)
+const libDir = (config.dest) ? path.resolve(`${__dirname}/../../../`, config.dest) : path.resolve(`${__dirname}/../lib`)
```

Comment:
What would be the use-case here? I cannot come up with a use-case where such a configuration would be useful. Maybe you have some specific case which you can share for me to better understand why we would need such a configuration.

Nevertheless, if there is no real use-case for such a scenario I would prefer to not implement this config for the following reasons:

1. If the user sets the `dest` config he/she cannot require the module anymore with `const sapui5 = require('sapui5-runtime')` as we return a hardcoded path to the lib folder. Take a look at [`index.js`](https://github.com/bastilimbach/sapui5-runtime/blob/master/src/index.js). If we want to support the config, we also need to adapt this file to return the correct path.
2. We might run into file authorization errors if the user provides a path to a restricted folder on his/her system
3. We don't check which path is entered so it is possible to mistakingly add relative paths like: `../../../../` which would lead to a security vulnerability. Also, the libDir is always deleted from the system before running the installation which would corrupt the file system of the end-user if such a relative path is entered.

Reply:
I understand your concerns. My purpose is to download the sapui5 runtime once and then have in a shared spot for all apps. I guess you could install the package globally instead maybe?
My usecase is to develop locally for all our apps 20+ and use one runtime instead of having to download it for every project.

Reply:
Ah ok, I get it.
This package is intended to be used as a local package and not a global one. For such a use-case I would either download the file manually and place it somewhere safe or write a plugin for the [UI5 tooling](https://github.com/SAP/ui5-tooling) which can be executed globally. This package should be used on a project by project basis as multiple UI5 projects typically require different SAPUI5 versions. What do you think?

Reply:
Yeah I understand your comment. I guess you could install it globally instead and just use the lib. I'll revert those changes.